File: 2022/Day 1/pt1.cpp
#include "helper.hpp"

int get_max_calories(std::vector<std::string> lines)
{
    int max_calories = 0;
    int current_calories = 0;

    for (std::string line : lines)
    {
        if (line == "")
        {
            if (current_calories > max_calories)
                max_calories = current_calories;
            current_calories = 0;
        }
        else
        {
            current_calories += std::atoi(line.c_str());
        }
    }

    return max_calories;
}

int main()
{
    std::vector<std::string> lines = Helper::get_lines("input.txt");

    auto t1 = Helper::get_current_time();
    int calories = get_max_calories(lines);
    std::cout << "Max Calories: " << calories << std::endl;
    auto t2 = Helper::get_current_time();
    Helper::get_time_taken(t1, t2);

    return 0;
}

File: 2023/Day 02/pt1.go
package main

import (
	"fmt"
	"regexp"
	"strconv"
	"strings"

	"github.com/benjababe/advent-of-code/helper"
)

func solve(lines []string) int {
	score := 0
	cubeMax := []int{12, 13, 14}

	for i, line := range lines {
		line = line[strings.Index(line, ":")+1:]
		possible := true
		patterns := []string{`(\d+)\sred`, `(\d+)\sgreen`, `(\d+)\sblue`}

		for colIndex, pattern := range patterns {
			regex := regexp.MustCompile(pattern)
			matches := regex.FindAllStringSubmatch(line, -1)

			for _, match := range matches {
				count, _ := strconv.Atoi(match[1])
				if count > cubeMax[colIndex] {
					possible = false
				}
			}
		}

		if possible {
			score += i + 1
		}
	}

	return score
}

func main() {
	lines := []string{}
	helper.GetLines(&lines, "input.txt")

	start := helper.GetCurrentTime()
	output := solve(lines)
	fmt.Printf("Output: %d\n", output)
	end := helper.GetCurrentTime()
	helper.GetTimeTaken(start, end)
}
